smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -271,6 +271,15 @@ contract MixinKeys is
     _keys[_tokenId].expirationTimestamp = newTimestamp;
 
     emit KeyExtended(_tokenId, newTimestamp);
+
+    // call the hook
+    if(address(onKeyExtendHook) != address(0)) {
```

Comment:
I think this will be called for `extend`, `grantKeyExtension` and `renew`, correct?

Reply:
yes, the three calls it to extend the time o the key